### PR TITLE
[rust-server] Added client documentation to rust-server

### PR DESCRIPTION
**Pull request by Herrera93**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh`, `./bin/security/{LANG}-petstore.sh` and `./bin/openapi3/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language. @frol @farcaller @bjgill

### Description of the PR

I added documentation for the client generated by `rust-server`. This documentation is based on [`rust` client docs](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator/src/main/resources/rust) with some modifications to fit the use of variables. 

I modified `RustServerCodegen` to include `String` as a primitive type and determine if a model property is a primitive, this way it won't show a reference to file if the property is primitive.

Some modifications that need to be done are:
 - Update the auth example
 - Find a way to determine if a nested container is primitive i.e. [`HashMap<String, HashMap<String, String>>`](
https://github.com/joylabs/openapi-generator/blob/a6ccc6bc4c17c81891d2fefe8ba55c19e3304fa6/samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing/docs/AdditionalPropertiesClass.md)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java [hunk 1]
```diff
@@ -783,7 +812,7 @@ public void postProcessParameter(CodegenParameter parameter) {
 
             String name = "models::" + getTypeDeclaration(parameter.dataType);
             parameter.dataType = name;
-            parameter.baseType = name;
+            // parameter.baseType = name;
```
Comment (bjgill) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java]: If this is unused, why not delete?

File: modules/openapi-generator/src/main/resources/rust-server/README.mustache [hunk 2]
```diff
@@ -110,3 +110,75 @@ Now replace the implementations in `src/server.rs` with your own code as require
...
+{{#isApiKey}}- **Type**: API key 
+
+Example
+```
+	auth := context.WithValue(context.TODO(), sw.ContextAPIKey, sw.APIKey{
+		Key: "APIKEY",
+		Prefix: "Bearer", // Omit if not necessary.
+	})
+    r, err := client.Service.Operation(auth, args)
+```
```
Comment (bjgill) [inline modules/openapi-generator/src/main/resources/rust-server/README.mustache]: This looks like go to me. Can we just delete it?

**Comment by Herrera93** [comment]
@bjgill Mainly There are doc comments but they only contain operation and model descriptions but do not contain parameters or properties descriptions.

I wanted to create centralized documentation for the client that can be explored in an interactive way so if there is a possibility that the user wants to expose the client.

**Comment by bjgill** [comment]
Ah, right. I see. In general, my preference would be to improve the doc comments rather than adding this parallel documentation. However, you've done this useful work, so I'll not let the perfect stand in the way of the good.

I'll give this a couple of days for other reviews, and then 🚢

**Comment by wing328** [comment]
@Herrera93 thanks for the PR 👍 

@bjgill thanks for the review.

Merging this into master to avoid too many outstanding PRs in this repo.

**Pull request merged**